src/controller/java/src/chip/devicecontroller/model/InvokeElement.java

```diff
@@ -19,18 +19,45 @@
 
 import java.util.Locale;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /** An invoke element that should be used for interaction model invoke request and response. */
 public final class InvokeElement {
   private final ChipPathId endpointId, clusterId, commandId;
-  private final byte[] tlv;
+  private final Optional<byte[]> tlv;
+  private final Optional<JSONObject> json;
+  private static Logger logger = Logger.getLogger(InvokeElement.class.getName());
 
   private InvokeElement(
-      ChipPathId endpointId, ChipPathId clusterId, ChipPathId commandId, byte[] tlv) {
+      ChipPathId endpointId,
+      ChipPathId clusterId,
+      ChipPathId commandId,
+      Optional<byte[]> tlv,
+      Optional<String> jsonString) {
     this.endpointId = endpointId;
     this.clusterId = clusterId;
     this.commandId = commandId;
-    this.tlv = tlv.clone();
+    if (tlv.isPresent()) {
```

Comment:
This can be simplified to:

```
this.tlv = tlv.map(data -> data.clone());
```

However, I'd recommend just storing the data as a `@Nullable` field rather than `Optional`. i.e. 

```
@Nullable private final byte[] tlv;

...

this.tlv = tlv.orElse(null);
```

Reply:
updated with nullable, thanks